src/Layer.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var Matrix = require('ml-matrix');
+
+var Utils = require('./Utils');
+const ACTIVATION_FUNCTIONS = require('./ActivationFunctions');
+
+class Layer {
+    /**
+     * Create a new layer with the given options
+     * @param {object} options
+     * @param {number} [options.inputSize] - Number of conections that enter the neurons.
+     * @param {number} [options.outputSize] - Number of conections that leave the neurons.
+     * @param {number} [options.regularization] - Regularization parameter.
+     * @param {number} [options.epsilon] - Learning rate parameter.
+     * @param {string} [options.activation] - Activation function parameter from the FeedForwardNeuralNetwork class.
+     * @param {number} [options.activationParam] - Activation parameter if needed.
+     */
+    constructor(options) {
+        this.inputSize = options.inputSize;
+        this.outputSize = options.outputSize;
+        this.regularization = options.regularization;
+        this.epsilon = options.epsilon;
+        this.activation = options.activation;
+        this.activationParam = options.activationParam;
+
+        var selectedFunction = ACTIVATION_FUNCTIONS[options.activation];
+        var params = selectedFunction.activation.length;
+
+        var actFunction = params > 1 ? val => selectedFunction.activation(val, options.activationParam) : selectedFunction.activation;
+        var derFunction = params > 1 ? val => selectedFunction.derivate(val, options.activationParam) : selectedFunction.derivate;
+
+        this.activationFunction = function (i, j) {
+            this[i][j] = actFunction(this[i][j]);
+        };
+        this.derivate = function (i, j) {
+            this[i][j] = derFunction(this[i][j]);
+        };
+
+        if (options.model) {
+            // load model
+            this.W = Matrix.checkMatrix(options.W);
+            this.b = Matrix.checkMatrix(options.b);
+
+        } else {
+            // default constructor
+
+            this.W = Matrix.rand(this.inputSize, this.outputSize);
+            this.b = Matrix.zeros(1, this.outputSize);
+
+            this.W.apply(function (i, j) {
+                this[i][j] /= Math.sqrt(options.inputSize);
+            });
+        }
+    }
+
+    /**
+     * propagate the given input through the current layer.
+     * @param {Matrix} X - input.
+     * @return {Matrix} output at the current layer.
+     */
+    forward(X) {
+        var z = X.mmul(this.W).addRowVector(this.b);
+        z.apply(this.activationFunction);
+        this.a = z.clone();
+        return z;
+    }
+
+    /**
+     * apply backpropagation algorithm at the current layer
+     * @param {Matrix} delta - delta values estimated at the following layer.
+     * @param {Matrix} a - 'a' values from the following layer.
+     * @return {Matrix} the new delta values for the next layer.
+     */
+    backpropagation(delta, a) {
+        this.dW = a.transpose().mmul(delta);
```

Comment:
You can win some time and memory by using a Matrix view for the transpose here and in other places. Just replace `a.transpose()` with `a.transposeView()`.
This is a relatively new feature. You can see how it works here: https://github.com/mljs/matrix/blob/master/src/views/transpose.js
And the list of currently implemented views: https://mljs.github.io/matrix/#Matrix#transposeView